### DISCUSSION TITLE
Recursive entities

### DISCUSSION THREAD

**Discussion by dmeehan1968** [tywalch/electrodb#418] [Q&A]
I'm looking for a solution to creating recursive entities, such as a group that has sub groups.

This can be modelled with attributes for `groupId` and `parentGroupId`, with a primary index that just uses groupId as a PK, to allow for direct retrieval of the group regardless of whether it has a parent.

It feels like I then need a secondary index with a collection, so that I can retrieve either the group alone, or the group and subgroups.  If I was doing this in Dynamo directly, I'd be able to conditionally define the secondary index, but in Electro it seems that although I can either define or not define the secondary, I can't control which composite keys go into it, so it can't be populated with the group itself at the 'root' of the partition.  `template` doesn't help as its only pattern matching, i.e. not able to expression conditional logic.

**Comment by tywalch** [comment]
Can you demonstrate more concretely what you're trying to accomplish? If you're able to accomplish this without Electro what does that look like?

**Reply by dmeehan1968** [reply]
Thanks for the speedy response, and on reflection I realise that I was getting confused by the 'collection' functionality of EDB.

Here is what I have been doing, and can model the same in EDB with a condition on the GSI to omit the index if the parentGroup is ''.

Primary PK groupId
GSI PK parentGroupId
GSK SK groupId

I thought somehow that collections let me put the group also in the GSI.
